File: 백준/Bronze/16769. Mixing Milk/Mixing Milk.py
import sys
input=lambda:sys.stdin.readline().rstrip()
c, m = [0 for i in range(3)], [0 for i in range(3)]
for i in range(3):
    c[i], m[i] = map(int, input().split())
for i in range(100):
    idx = i % 3
    next_idx = (idx + 1) % 3
    m[idx], m[next_idx] = max(m[idx] - (c[next_idx] - m[next_idx]), 0), min(c[next_idx], m[idx] + m[next_idx])
for num in m:
    print(num)

File: 백준/Bronze/16769. Mixing Milk/Mixing Milk.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int idx, nidx, t;
    vector<vector<int>> arr(3, vector<int>(2, 0));
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 2; j++)
            cin >> arr[i][j];
    }
    for (int i = 0; i < 100; i++) {
        idx = i % 3, nidx = (i + 1) % 3;
        t = arr[idx][1];
        arr[idx][1] = max(arr[idx][1] - (arr[nidx][0] - arr[nidx][1]), 0);
        arr[nidx][1] = min(arr[nidx][0], t + arr[nidx][1]);
    }
    for (int i = 0; i < 3; i++)
        cout << arr[i][1] << "\n";
    return 0;
}